enhancements/update/to-multi-arch.md

```diff
@@ -0,0 +1,157 @@
+  // +optional
+  Architecture string `json:"architecture"`
+}
+```
+
+and [`ClusterVersion.status.desired`](https://github.com/openshift/api/blob/82e082220d910f489d35880174e2eb90e21f5589/config/v1/types_cluster_version.go#L741) with:
+
+
+```go
+type Release struct {
+	// Architecture is the architecture for which the release image is built, such as "amd64", "arm64", "ppc64le", "s390x" "multi", etc.
+	// +optional
+	Architecture string `json:"architecture"`
+  ...
+}
+```
+
+### Topology Considerations
+
+TODO (hongkliu): I have no idea about how CVO works differently from Standalone Clusters.
+
+### Implementation Details/Notes/Constraints
+
+#### Operator's deployment
+
+Each operator's deployment has an environment variable `OPERATOR_IMAGE_ARCHITECTURE` for the architecture of the operator image 
+and `OPERAND_NAME_IMAGE_ARCHITECTURE` for the architecture of each operand that the operator manages. The values are vendored by ART that builds the OpenShift releases with the `oc adm release new` command.
+
+```
+env:
+- name: OPERATOR_IMAGE_ARCHITECTURE
+  value: UNDEFINED-ARCHITECTURE
+```
+
+Each operator takes the values of those environment variables to fill in the `architecture` field in `ClusterOperator.status.versions`.
+The default value is the empty string if any of the environment variable is not defined so that operator can use the new field in the API without waiting for ART to actually define them.
+
+A special operator: [machine-config-operator](https://github.com/openshift/machine-config-operator) (MCO).
+It is special because it is the last operator to upgrade in a cluster upgrade and it usually takes longest for MCO to upgrade. With its adoption of the extended API, the ending point of migrating to multi-arch will become accurate which improves the overall UX the most.
+
+The `oc adm release new` command will be modified to render the environment variables such as `OPERATOR_IMAGE_ARCHITECTURE` with expected values by replacing "UNDEFINED-ARCHITECTURE" with "amd64", "arm64", "multi", etc. See how the "versions" are [handled in the command](https://github.com/openshift/oc/blob/master/pkg/cli/admin/release/image_mapper.go#L342-L348).
+
+
+A simple implementation is to ask ART to use `--metadata {"release.openshift.io/architecture": "<ARCHITECTURE>"}` in the command for all the cases. ART uses it only for "multi" today. Then `<ARCHITECTURE>` there or the empty string as the default value is used to replace "UNDEFINED-ARCHITECTURE" in the deployment manifest for each operator.
+
+We will allows [other values than "Multi" and ""](https://github.com/openshift/api/blob/a1523024209f3122be9e9d53515da470c6a2458b/config/v1/types_cluster_version.go#L281) and modify the code [here](https://github.com/openshift/cluster-version-operator/blob/455abd6acc3e69653e7506b1cef4cbdcd2512b92/pkg/payload/payload.go#L388) in CVO to accept them.
```

Comment:
Or we could ask "operators" to do the trick as CVO does today: env-var is only for multi and runtime for others.
We may do the trick in go-library.
Thus, ART does not to do anything and we do not need to modify the API about this part.

Reply:
Not relevant since we do not change API any more.